apps/utils/cipher/cipher.main.kts

```diff
@@ -0,0 +1,504 @@
+        
+          ${Span.Secure.start.text}...any text you would like to have encrypted...${Span.Secure.end.text}
+        
+        Cipher blocks are created and maintained by the script.
+        
+          ${Span.Cipher.start.text}...base 64 encrypted data from a secure block...${Span.Cipher.end.text}
+    """.trimIndent()
+    )
+}
+
+data class Config(
+    val key: String,
+    val endpoint: String?,
+    val region: String,
+    val mode: String,
+    val input: String,
+    val output: String,
+    val editor: String
+)
+
+fun parseArgs(args: Array<String>): Config {
+    var key: String? = System.getenv("CIPHER_KEY")
+    var endpoint: String? = System.getenv("CIPHER_ENDPOINT")
+    var region: String = System.getenv("CIPHER_REGION") ?: DEFAULT_REGION
+    val editor = System.getenv("EDITOR") ?: DEFAULT_EDITOR
+
+    var i = 0
+    while (i < args.size && args[i].startsWith("--")) {
+        when (args[i]) {
+            "--key" -> {
+                i += 1
+                key = args[i]
+            }
+
+            "--endpoint" -> {
+                i += 1
+                endpoint = args[i]
+            }
+
+            "--region" -> {
+                i += 1
+                region = args[i]
+            }
+
+            else -> {
+                throw UsageException("invalid option ${args[i]}")
+            }
+        }
+        i += 1
+    }
+
+    if (key == null) {
+        throw UsageException("required --key or CIPHER_KEY not provided")
+    }
+
+    val remaining = args.size - i
+    if (remaining < 2 || remaining > 3) {
+        throw UsageException("expected 2 or 3 arguments")
+    }
+
+    val mode = args[i++]
+    val input = args[i++]
+    val output = if (i < args.size) args[i] else input
+
+    return Config(key, endpoint, region, mode, input, output, editor)
+}
+
+fun main(args: Array<String>) {
+    val config = parseArgs(args)
+    val cipher = KmsCipher(config.endpoint, config.region, config.key)
+    val editor = config.editor
+
+    val mode = config.mode
+    val inputFile = File(config.input)
+    val outputFile = File(config.output)
+    when (mode) {
+        "cat" -> {
+            val original = loadText(inputFile)
+            val decrypted = original.decrypt(cipher)
+            val printBuffer = Buffer()
+            decrypted.store(printBuffer)
+            print(printBuffer.readUtf8())
+        }
+
+        "encrypt" -> {
+            val original = loadText(inputFile)
+            val encrypted = original.encrypt(cipher)
+            encrypted.store(outputFile.sink())
+        }
+
+        "decrypt" -> {
+            val original = loadText(inputFile)
+            val decrypted = original.decrypt(cipher)
+            decrypted.store(outputFile.sink())
+        }
+
+        "rewind" -> {
+            val original = loadText(inputFile)
+            val rewound = original.rewind(cipher)
+            rewound.store(outputFile.sink())
+        }
+
+        "edit" -> {
```

Comment:
Yes, edit could use some attention for this and other reasons.  Maybe this could be added in a follow-on sprint.  I'm sure we'll gain a lot of insights through actual experience with the tool.

Reply:
I agree. I don't think it's possible that we will collectively lose sight of usability challenges when we actually start using this, right? With that, I think we can move ahead even without opening a ticket at this time.